notification/search.go

```diff
@@ -123,6 +143,10 @@ type renderData struct {
 	TimeSeries         bool
 	TimeSeriesOrigin   time.Time
 	TimeSeriesInterval time.Duration
+
+	SegmentByService     bool
+	SegmentByUser        bool
+	SegmentByMessageType bool
```

Comment:
Can we define these as methods instead that just check `SearchOptions`? that way we're not duplicating data (i.e., which is source of truth, these books or the value in SearchOptions)